From 936199a93f500516dc599293f5c4cf6ffb7aab1d Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Mon, 12 Mar 2018 23:38:08 +0530
Subject: [PATCH 0332/1852] drm: xlnx: scaler: Updating vpss-scaler reset logic

commit 0f60df6f0625a7b57b9a05a44097b6a516376674 from
https://github.com/Xilinx/linux-xlnx.git

This patch updates the reset through gpio logic

Signed-off-by: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_scaler.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_scaler.c b/drivers/gpu/drm/xlnx/xlnx_scaler.c
index 8460b1691292..d96ddeb49b59 100644
--- a/drivers/gpu/drm/xlnx/xlnx_scaler.c
+++ b/drivers/gpu/drm/xlnx/xlnx_scaler.c
@@ -1525,8 +1525,8 @@ static int xilinx_scaler_bridge_set_input(struct xlnx_bridge *bridge,
 	scaler->fmt_in = bus_fmt;
 
 	/* IP Reset through GPIO */
-	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_DEASSERT);
 	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_ASSERT);
+	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_DEASSERT);
 	xilinx_scaler_reset(scaler);
 	memset(scaler->H_phases, 0, sizeof(scaler->H_phases));
 
@@ -1634,7 +1634,7 @@ static int xilinx_scaler_probe(struct platform_device *pdev)
 	scaler->max_pixels = XSCALER_MAX_WIDTH;
 
 	/* Reset the Global IP Reset through a GPIO */
-	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_ASSERT);
+	gpiod_set_value_cansleep(scaler->rst_gpio, XSCALER_RESET_DEASSERT);
 	xilinx_scaler_reset(scaler);
 
 	scaler->bridge.enable = &xilinx_scaler_bridge_enable;
-- 
2.31.1

